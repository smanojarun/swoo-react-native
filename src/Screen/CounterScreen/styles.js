import { StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
    flex1: {
        flex: 1
    },
    container: {
        width: "100%",
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
    },
    button: {
        paddingVertical: 12,
        paddingHorizontal: 50,
        borderRadius: 20,
    },
    btnText: {
        fontSize: 16,
        fontWeight: '500'
    },
    valueTxt: {
        paddingVertical: 90,
        fontSize: 46,
        fontWeight: 'bold',
    },
});
