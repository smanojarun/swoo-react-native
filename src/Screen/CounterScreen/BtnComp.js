import React from "react"
import { TouchableOpacity, Text } from "react-native"
import { styles } from "./styles"

const BtnComp = (props) => {
    const { compID, buttonColor, labelColor, label, 
        onPress } = props

    return (
        <TouchableOpacity testID={compID} style={[styles.button, { backgroundColor: buttonColor }]} onPress={() => props.onPress()}>
            <Text style={[styles.btnText, { color: labelColor }]}>{label}</Text>
        </TouchableOpacity>
    )
}

export default BtnComp