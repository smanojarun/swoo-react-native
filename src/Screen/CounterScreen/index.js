import React, { useContext } from 'react';
import {
  SafeAreaView,
  StatusBar,
  Text,
  TouchableOpacity,
  useColorScheme,
  View,
} from 'react-native';
import { connect } from 'react-redux';
import MyContext from '../../context/MyContext';
import { decrementCounter, incrementCounter } from '../../reducer/actions';
import { styles } from './styles'
import BtnComp from './BtnComp'

const CounterScreen = (props) => {
  const isDarkMode = useColorScheme() === 'dark';
  const { themeColor } = useContext(MyContext)

  const doIncrement = () => {
    props.incrementCounter()
  }

  const doDecrement = () => {
    props.decrementCounter()
  }

  return (
    <SafeAreaView style={styles.flex1}>
      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
      <View style={[styles.container, { backgroundColor: themeColor.background }]}>
        <BtnComp compID="inc" buttonColor={themeColor.button}  labelColor={themeColor.btnText} onPress={() => doIncrement()} label={"+1"} />
        <Text style={[styles.valueTxt, { color: themeColor.valueText }]}>{props.data.counterValue}</Text>
        <BtnComp compID="dec" buttonColor={themeColor.button} labelColor={themeColor.btnText} onPress={() => doDecrement()} label={"-1"} />
      </View>
    </SafeAreaView>
  );
};


const mapStateToProps = (state) => {
  return {
    data: state.counterReducer
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    incrementCounter: () => { dispatch(incrementCounter()) },
    decrementCounter: () => { dispatch(decrementCounter()) }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CounterScreen);
