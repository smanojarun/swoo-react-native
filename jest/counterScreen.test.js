import 'react-native';
import React from 'react';
import Root from '../App'
import CounterScreen from '../src/Screen/CounterScreen';
import { shallow, mount } from 'enzyme';
import BtnComp from '../src/Screen/CounterScreen/BtnComp';

describe("Root Component", () => {
    const wrapper = shallow(
        <Root />
    );

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    });

    it('should Test for increment button to be present', () => {
        expect(wrapper
            .find('TouchableOpacity')
            .findWhere((node) => node.prop('testID') == 'inc')).toBeTruthy()
    })

    it('should Test for decrement button to be present', () => {
        expect(wrapper
            .find('TouchableOpacity')
            .findWhere((node) => node.prop('testID') == 'dec')).toBeTruthy()
    })

    it('should Test for Increment button Click Action', () => {
        const onPress = jest.fn()
        const component = shallow(<BtnComp compID={'inc'} onPress={onPress} />)
        component.find('TouchableOpacity')
            .findWhere((node) => node.prop('testID') == 'inc').props().onPress()
        expect(onPress).toHaveBeenCalledTimes(1);
    })

    it('should Test for Decrement button Click Action', () => {
        const onPress = jest.fn()
        const component = shallow(<BtnComp compID={'dec'} onPress={onPress} />)
        component.find('TouchableOpacity')
            .findWhere((node) => node.prop('testID') == 'dec').props().onPress()
        expect(onPress).toHaveBeenCalledTimes(1);
    })
});