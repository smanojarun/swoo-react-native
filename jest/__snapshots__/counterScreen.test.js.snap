// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component renders correctly 1`] = `ShallowWrapper {}`;

exports[`Login Component renders correctly 1`] = `ShallowWrapper {}`;

exports[`Root Component renders correctly 1`] = `ShallowWrapper {}`;

exports[`renders correctly 1`] = `ShallowWrapper {}`;
